test/images/image-util.sh

```diff
@@ -49,57 +75,78 @@ build() {
-    if [[ -f ${IMAGE}/Makefile ]]; then
-      # make bin will take care of all the prerequisites needed
-      # for building the docker image
-      make -C "${IMAGE}" bin ARCH="${arch}" TARGET="${temp_dir}"
-    fi
-    pushd "${temp_dir}"
-    # image tag
-    TAG=$(<VERSION)
-
-    if [[ -f BASEIMAGE ]]; then
-      BASEIMAGE=$(getBaseImage "${arch}")
-      ${SED} -i "s|BASEIMAGE|${BASEIMAGE}|g" Dockerfile
-      ${SED} -i "s|BASEARCH|${arch}|g" Dockerfile
-    fi
-
-    # copy the qemu-*-static binary to docker image to build the multi architecture image on x86 platform
-    if grep -q "CROSS_BUILD_" Dockerfile; then
-      if [[ "${arch}" == "amd64" ]]; then
-        ${SED} -i "/CROSS_BUILD_/d" Dockerfile
-      else
-        ${SED} -i "s|QEMUARCH|${QEMUARCHS[$arch]}|g" Dockerfile
-        # Register qemu-*-static for all supported processors except the current one
-        echo "Registering qemu-*-static binaries in the kernel"
-        local sudo=""
-        if [[ $(id -u) != 0 ]]; then
-          sudo=sudo
+  case "$(uname -m)" in
+    x86_64|amd64)
+      for arch in ${archs}; do
+        prepareBuild "${arch}"
+        # image tag
+        TAG=$(<VERSION)
+        # copy the qemu-*-static binary to docker image to build the multi architecture image on x86 platform
+        if grep -q "CROSS_BUILD_" Dockerfile; then
+          if [[ "${arch}" == "amd64" ]]; then
+            ${SED} -i "/CROSS_BUILD_/d" Dockerfile
+          else
+            ${SED} -i "s|QEMUARCH|${QEMUARCHS[$arch]}|g" Dockerfile
+            # Register qemu-*-static for all supported processors except the current one
+            echo "Registering qemu-*-static binaries in the kernel"
+            local sudo=""
+            if [[ $(id -u) != 0 ]]; then
+              sudo=sudo
+            fi
+            ${sudo} "${KUBE_ROOT}/third_party/multiarch/qemu-user-static/register/register.sh" --reset
+            curl -sSL https://github.com/multiarch/qemu-user-static/releases/download/"${QEMUVERSION}"/x86_64_qemu-"${QEMUARCHS[$arch]}"-static.tar.gz | tar -xz -C "${temp_dir}"
+            # Ensure we don't get surprised by umask settings
+            chmod 0755 "${temp_dir}/qemu-${QEMUARCHS[$arch]}-static"
+            ${SED} -i "s/CROSS_BUILD_//g" Dockerfile
+          fi
         fi
-        ${sudo} "${KUBE_ROOT}/third_party/multiarch/qemu-user-static/register/register.sh" --reset
-        curl -sSL https://github.com/multiarch/qemu-user-static/releases/download/"${QEMUVERSION}"/x86_64_qemu-"${QEMUARCHS[$arch]}"-static.tar.gz | tar -xz -C "${temp_dir}"
-        # Ensure we don't get surprised by umask settings
-        chmod 0755 "${temp_dir}/qemu-${QEMUARCHS[$arch]}-static"
-        ${SED} -i "s/CROSS_BUILD_//g" Dockerfile
-      fi
-    fi
-
-    docker build --pull -t "${REGISTRY}/${IMAGE}-${arch}:${TAG}" .
-
-    popd
-  done
+        docker build --pull -t "${REGISTRY}/${IMAGE}-${arch}:${TAG}" .
+        popd
+      done
+      ;;
+    arm*|aarch*)
```

Comment:
All the non-amd64 code is same, I'm wondering whether we club all non-amd64 code into one!

Reply:
agreed!

Reply:
consolidated some command code into a separate method.